addons/web/controllers/main.py

```diff
@@ -325,18 +325,29 @@ def set_cookie_and_redirect(redirect_url):
     return redirect
 
 def clean_action(action, env):
-    action.setdefault('flags', {})
     action_type = action.setdefault('type', 'ir.actions.act_window_close')
     if action_type == 'ir.actions.act_window':
         action = fix_view_modes(action)
 
     # When returning an action, only few information are really usefull
```

Comment:
while we're updating the logic, might as well fix comments (_useful_ takes only one L anyway)

```suggestion
    # When returning an action, keep only relevant fields/properties
```